internal/bminventory/inventory.go

```diff
@@ -1668,6 +1676,107 @@ func (b *bareMetalInventory) UpdateClusterNonInteractive(ctx context.Context, pa
+		return nil, nil, err
+	}
+
+	if *cluster.Platform.Type == models.PlatformTypeBaremetal {
+
+		// case 2
+		if !swag.BoolValue(userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeBaremetal) {
+			return nil, nil, nil
+		}
+
+		// case 3
+		if (platform != nil && *platform.Type == models.PlatformTypeNone) || (swag.BoolValue(userManagedNetworking) && platform == nil) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
+	}
+
+	if *cluster.Platform.Type == models.PlatformTypeNone {
+
+		// case 4
+		if (userManagedNetworking == nil || *userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeNone) {
+			return nil, nil, nil
+		}
+
+		// case 6
+		if !swag.BoolValue(userManagedNetworking) && (platform == nil || *platform.Type == models.PlatformTypeBaremetal) {
+			if cluster.CPUArchitecture != common.X86CPUArchitecture && !featuresupport.IsFeatureSupported(cluster.OpenshiftVersion, models.FeatureSupportLevelFeaturesItems0FeatureIDARM64ARCHITECTUREWITHCLUSTERMANAGEDNETWORKING) {
+				return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed for clusters with non-x86_64 CPU architecture"))
+			}
+
+			if *cluster.HighAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+				return nil, nil, common.NewApiError(http.StatusBadRequest, errors.New("disabling User Managed Networking or setting Bare-Metal platform is not allowed in single node mode"))
+			}
+
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+		}
+	}
+
+	return platform, userManagedNetworking, nil
+}
+
+func getActualCreateClusterPlatformParams(platform *models.Platform, userManagedNetworking *bool, highAvailabilityMode *string) (*models.Platform, *bool, error) {
+
+	if err := checkPlatformWrongParamsInput(platform, userManagedNetworking); err != nil {
+		return nil, nil, err
+	}
+
+	if *highAvailabilityMode == models.ClusterHighAvailabilityModeFull {
+		if (platform == nil || *platform.Type == models.PlatformTypeBaremetal) && !swag.BoolValue(userManagedNetworking) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeBaremetal)}, swag.Bool(false), nil
+		}
+
+		if swag.BoolValue(userManagedNetworking) || (platform != nil && *platform.Type == models.PlatformTypeNone) {
+			return &models.Platform{Type: common.PlatformTypePtr(models.PlatformTypeNone)}, swag.Bool(true), nil
+		}
+	}
+
+	if *highAvailabilityMode == models.ClusterHighAvailabilityModeNone {
+		if platform != nil && *platform.Type == models.PlatformTypeBaremetal && userManagedNetworking != nil && !*userManagedNetworking {
+			return nil, nil, common.NewApiError(http.StatusBadRequest,
+				errors.Errorf("Can't set %s platform and disable user-managed-networking on single node OpenShift", models.PlatformTypeBaremetal))
+		}
```

Comment:
No need for this statement (and since it's complicated as it is I think we can remove it), the user will get  badRequest from the next 2 if statements until if he try to set platform=Baremetal or userManagedNetworking=True.

Reply:
The message here is clearer, but not sure it's worth the extra code.  I agree, can remove.